### PR TITLE
[SIDE-477] Documentation Updates

### PR DESCRIPTION
**Pull request by richseviora**
This PR:

- Replaces JSDoc with TypeDoc.
- Updates all the corresponding documentation:
  - Removes JSDoc specific tags and type annotations.
  - Marks internal classes and interfaces so they're not visible in the generated documentation.
- Creates interfaces for the Controller and Sensor implementations.
- Creates interfaces for the host objects received by the Controllers and Sensors, and moved up the common storage methods into a shared `PluginHost` interface.
 - Restructures the readme and adds in specific steps to take.

Screen Shot 2020-08-18 at 11 03 01 AM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,6 +50,8 @@
     "lint-staged": "10.2.11",
     "pkg": "latest",
     "prettier": "2.0.5",
+    "typedoc": "0.17.0-3",
```
Comment (BillyMurrayOlive) [inline package.json:53]: I think we can remove the jsdoc dependency on line 49.

File: readme.md [hunk 2]
```diff
@@ -1,16 +1,16 @@
 # Loop Development Kit (LDK) for NodeJS
+## Developing
+### Setup
+#### Setting Up Your Plugin
```
Comment (BillyMurrayOlive) [inline readme.md:4]: This is great! Much more direct.

File: src/sensorHost.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { PluginEvent } from './pluginEvent';
+import { PluginHost } from './pluginHost';
+
+export interface SensorHost extends PluginHost {
```
Comment (BillyMurrayOlive) [inline src/sensorHost.ts:4]: I think this is a good pattern you have setup here.

File: package.json (hunk 1 above)
Comment (richseviora) [reply package.json:51]: Probably 😄  Is done

**Pull request merged**